ngraph/frontend/generic/include/frontend_manager/frontend_manager.hpp

```diff
@@ -0,0 +1,454 @@
+        /// retrieve instances of Place to point to particular node part which should be modified or
+        /// satisfies some criteria. For example, this class is used to report model inputs
+        /// and outputs, for searching operations and tensors by name, for setting shape etc.
+        ///
+        /// Place can refer to Tensor, Input Edge, Input Port, Operation, Output Port, Output Edge
+        ///
+        ///                [Tensor A]
+        ///                    |
+        ///                    | [Input Edge]
+        ///                    |
+        ///                    V
+        ///           -------------------
+        ///           [  [Input Port 0] ]
+        ///           [                 ]
+        ///           [   Operation A   ]
+        ///           [                 ]
+        ///           [ [Output Port 0] ]
+        ///           -------------------
+        ///                    |
+        ///                    | [Output Edge]
+        ///                    |
+        ///                    V
+        ///                [Tensor B]
+        ///                    |
+        ///                    | [Input Edge]
+        ///                    |
+        ///                    V
+        ///           -------------------
+        ///           [  [Input Port 0] ]
+        ///           [                 ]
+        ///           [   Operation B   ]
+        ///           [                 ]
+        ///           [ [Output Port 0] ]
+        ///           -------------------
+        ///                    |
+        ///                    | [Output Edge]
+        ///                    |
+        ///                    V
+        ///                [Tensor C]
+        ///
+        class FRONTEND_API Place
+        {
+        public:
+            typedef std::shared_ptr<Place> Ptr;
+
+            virtual ~Place() = default;
+
+            /// \brief All associated names (synonyms) that identify this place in the graph in a
+            /// framework specific way
+            ///
+            /// \return A vector of strings each representing a name that identifies this place in
+            /// the graph. Can be empty if there are no names associated with this place or name
+            /// cannot be attached.
+            virtual std::vector<std::string> get_names() const;
+
+            /// \brief Returns references to all operation nodes that consume data from this place
+            /// \note It can be called for any kind of graph place searching for the first consuming
+            /// operations.
+            ///
+            /// \param outputPortIndex If place is an operational node it specifies which output
+            /// port should be considered
+            ///
+            /// \return A vector with all operation node references that consumes data from this
+            /// place
+            virtual std::vector<Ptr> get_consuming_operations(int outputPortIndex = -1) const;
```

Comment:
What does the -1 value mean?

Reply:
-1 was a temporary solution for case when an operation has only a single output port and this method takes it by default, it should throw if there are multiple outputs (even not connected). If we want to be aligned with `ngraph::Node` then we have to remove this capability because there we request explicitly specified index. @GlebKazantaev what is your opinion?
If we want to keep this capability, then it should be a separate method without outputPortIndex.